### ISSUE TITLE
[FEAT] how to deal with categorical features in TimeGAN? does TimeGAN converge? 

### ISSUE THREAD

**Issue by liqi6811** [ydataai/ydata-synthetic#89]
Hi team, 

This project is excellent, it makes much better sense to me as a tf2 person. Appreciate your work and sharing!! 

I have two questions: 
1) If my data has categorical data, may I still use the same architecture to generate synthetic time series data? In function read_data_loading you used MinMaxScaler() to scale continuous data, if I have categorical data, should I just do one-hot encoding and concatenate with scaled numerical data? 

2) I notice discrimination loss is using binary cross entropy loss function, does this mean TimeGAN does not converge as the original GAN? I noticed you can save generator when TimeGAN finishes training, if it does not converge, should we save at evey n epochs and find out the best performing model? 

Best regards, 
Ling

**Comment by fabclmnt** [comment]
Hi @liqi6811,

great questions! Let me see if I can assist you:

1.  For you to be able to use new datasets a new processing or transformations methods are recommended to be develop by you. Those preprocessors should aim the characteristics of your data, in the case of categorical variables one-hot encoding might be suitable depending on the cardinality. 
Nevertheless, for more optimal results changes to the architecture might be needed, for example use Gumbel-Softmax: https://towardsdatascience.com/what-is-gumbel-softmax-7f6d9cdcb90e

2. Not sure what do you mean with this point.

**Comment by fabclmnt** [comment]
I'll be closing this issue. Any further questions let me know I'll be happy to answer you through our community slack - https://syntheticdata.community/.
